### PR TITLE
Build app.so instead of app.a on iOS

### PR DESCRIPTION
**Pull request by abarth**
The app.a wasn't getting pulled into the main executable because we weren't
referencing any of its symbols. Instead, create a dylib that can be packaged
with the application and loaded at runtime.

### CODE REVIEW

**Comment by abarth** [comment]
@chinmaygarde

File: packages/flutter_tools/lib/src/commands/build_aot.dart [hunk 1]
```diff
@@ -100,8 +100,12 @@ String buildAotSnapshot(
   } else {
     String artifactsDir = tools.getEngineArtifactsDirectory(platform, buildMode).path;
     entryPointsDir = artifactsDir;
-    String hostToolsDir = path.join(artifactsDir, getNameForHostPlatform(getCurrentHostPlatform()));
-    genSnapshot = path.join(hostToolsDir, 'gen_snapshot');
+    if (platform == TargetPlatform.ios) {
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/build_aot.dart]: btw the dream is to have all this platform-specific logic abstracted out into virtual methods on the platform objects so you just invoke the right method on the platform object and the right thing happens. Ideally we'd never check the platform type in non-platform-specific code.

**Comment by chinmaygarde** [comment]
lgtm

File: packages/flutter_tools/lib/src/commands/build_aot.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter_tools/lib/src/commands/build_aot.dart]: Yeah, this part got a little messy.  Once I get this working, I'll come back and refactor.

**Pull request merged**